### PR TITLE
Add rms norm[SiliconFlow]

### PR DESCRIPTION
**Pull request by MARD1NO**

### CODE REVIEW

File: src/flag_gems/__init__.py [hunk 1]
```diff
@@ -23,6 +23,8 @@ def enable(lib=aten_lib):
     lib.impl("isinf", isinf, "CUDA")
     lib.impl("isnan", isnan, "CUDA")
     lib.impl("native_layer_norm", layer_norm, "AutogradCUDA")
+    lib.impl("rms_norm", rms_norm, "AutogradCUDA")
```
Comment (StrongSpoon) [inline src/flag_gems/__init__.py]: register with "CUDA"

File: tests/flag_gems/op_accu_test.py [hunk 2]
```diff
@@ -523,6 +523,43 @@ def test_accuracy_layernorm(shape, dtype):
...
+)
+@pytest.mark.parametrize("dtype", [torch.float16, torch.float32, torch.bfloat16])
+def test_accuracy_rmsnorm(shape, dtype):
+    M = shape[0]
+    N = shape[1]
+    layer_shape = [
+        N,
+    ]
+    inp = torch.randn(shape, dtype=dtype, device="cuda", requires_grad=True)
+    weight = torch.randn(layer_shape, dtype=dtype, device="cuda", requires_grad=True)
```
Comment (StrongSpoon) [inline tests/flag_gems/op_accu_test.py]: test of forward function doesn't require grad

File: src/flag_gems/ops/rms_norm.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+):
+    pid = tl.program_id(0)
+    Y += pid * y_stride_r
+    X += pid * x_stride_r
+
+    mask = tl.arange(0, BLOCK_SIZE) < N
+    cols = tl.arange(0, BLOCK_SIZE)
+    x = tl.load(X + cols * x_stride_c, mask, other=0.0).to(tl.float32)
+
+    var = tl.sum(x * x, axis=0) / N
```
Comment (StrongSpoon) [inline src/flag_gems/ops/rms_norm.py]: I prefer sum(x*x/N)

File: src/flag_gems/ops/rms_norm.py [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+        BLOCK_SIZE = triton.next_power_of_2(N)
+        x = x.contiguous()
+        weight = weight.contiguous()
+        y = torch.empty_like(x)
+
+        rms_norm_kernel[M, ](y, x, weight, N, 1, N, 1, N, eps, BLOCK_SIZE)
+        return y
+
+    @staticmethod
+    def backward(ctx, out_grad):
```
Comment (StrongSpoon) [inline src/flag_gems/ops/rms_norm.py]: remove backward till it's implemented

Comment (MARD1NO) [reply src/flag_gems/ops/rms_norm.py]: Done

File: src/flag_gems/ops/rms_norm.py (hunk 3 above)
Comment (MARD1NO) [reply src/flag_gems/ops/rms_norm.py]: Done

File: tests/flag_gems/op_accu_test.py (hunk 2 above)
Comment (MARD1NO) [reply tests/flag_gems/op_accu_test.py]: Done

File: src/flag_gems/__init__.py (hunk 1 above)
Comment (MARD1NO) [reply src/flag_gems/__init__.py]: Done

**Pull request merged**